stonesoup/platform/base.py

```diff
@@ -1,122 +1,83 @@
 
-        """
-    states: Sequence[State] = Property(
-        doc="A list of States which enables the platform's history to be "
-            "accessed in simulators and for plotting. Initiated as a "
-            "state, for a static platform, this would usually contain its "
-            "position coordinates in the form ``[x, y, z]``. For a moving "
-            "platform it would contain position and velocity interleaved: "
-            "``[x, vx, y, vy, z, vz]``")
-    position_mapping: Sequence[int] = Property(
-        doc="Mapping between platform position and state vector. For a "
-            "position-only 3d platform this might be ``[0, 1, 2]``. For a "
-            "position and velocity platform: ``[0, 2, 4]``")
-    rotation_offsets: MutableSequence[StateVector] = Property(
-        default=None, readonly=True,
-        doc="A list of StateVectors containing the sensor rotation "
-            "offsets from the platform's primary axis (defined as the "
-            "direction of motion). Defaults to a zero vector with the "
-            "same length as the Platform's :attr:`position_mapping`")
-    mounting_offsets: MutableSequence[StateVector] = Property(
-        default=None, readonly=True,
-        doc="A list of StateVectors containing the sensor translation "
-            "offsets from the platform's reference point. Defaults to "
-            "a zero vector with the same length as the Platform's "
-            ":attr:`position_mapping`")
-    sensors: MutableSequence['BaseSensor'] = Property(
+    .. note:: This class is abstract and not intended to be instantiated. To get the behaviour
+        of this class use a subclass which gives movement
+        behaviours. Currently these are :class:`~.FixedPlatform` and
+        :class:`~.MovingPlatform`
+    """
+
+    movement_controller: Movable = Property(
+        default=None,
+        doc=":class:`~.Movable` object to control the Platform's movement. Default is None, but "
+            "it can be constructed transparently by passing Movable's constructor parameters to "
+            "the Platform constructor.")
+    sensors: MutableSequence[Sensor] = Property(
         default=None, readonly=True,
         doc="A list of N mounted sensors. Defaults to an empty list")
-    velocity_mapping: Sequence[int] = Property(
-        default=None,
-        doc="Mapping between platform velocity and state dims. If not "
-            "set, it will default to ``[m+1 for m in position_mapping]``")
 
-    # TODO: Determine where a platform coordinate frame should be maintained
+    _default_movable_class = None  # Will be overridden by subclasses
+
+    def __getattribute__(self, name):
+        # This method is called if we try to access an attribute of self. First we try to get the
+        # attribute directly, but if that fails, we want to try getting the same attribute from
+        # self.state instead. If that, in turn,  fails we want to return the error message that
```

Comment:
Attempt to get attribute from the `controller` attribute, not the state

Reply:
Documentation corrected